src/backend/storage/file/buffile.c

```diff
@@ -164,6 +164,11 @@ static void BufFileDumpCompressedBuffer(BufFile *file, const void *buffer, Size
 static void BufFileEndCompression(BufFile *file);
 static int BufFileLoadCompressedBuffer(BufFile *file, void *buffer, size_t bufsize);
 
+#ifdef HAVE_LIBZSTD
+static void *customAlloc(void *opaque, size_t size);
```

Comment:
Prototypes for static functions are redundant if the function is defined before its first use.

Reply:
It is, but gpdb defines them even if they are not used before they are implemented, so I just used the current style